quinn/src/streams.rs

```diff
@@ -431,6 +436,46 @@ where
         }
     }
 
+    /// Read the next segment of data from next valid offset in the stream.
+    ///
+    /// Yields a segment of data consecutive to the last ordered read offset, or `None`
+    /// if the stream was finished. Segments are received in order.
+    ///
+    /// Chunk reads have reduced overhead and higher throughput, and should therefore be
+    /// preferred when applicable.
+    fn poll_read_chunk(&mut self, cx: &mut Context) -> Poll<Result<Option<Bytes>, ReadError>> {
```

Comment:
Also pretty repetitive. Can we factor these things?

Reply:
Would a macro be acceptable here ?

Reply:
This one wasn't that hard to refactor with a generic `fn`, actually.